jobs/ssh/regenerate_puppet_certificate.erb

```diff
@@ -0,0 +1,25 @@
+<%#
+name: Regenerate puppet certificate
+job_category: Puppet
+description_format: Regenerate puppet certificate
+snippet: false
+provider_type: SSH
+kind: job_template
+%>
+
+<% # this template requires safe mode to be disabled -%>
+
+<%= render_template('Service Action - SSH Default', :service => 'puppet', :action => 'stop') -%>
+
+
+<% if preview? -%>
+#   this would delete a certificate on puppet ca, but running in preview mode
+<% else -%>
+  <%= @host.puppet_ca_proxy.statuses[:puppetca].destroy(@host.name) && '# deleted cert on proxy' %>
+<% end -%>
+# delete the old cert and key
+rm -rf /var/lib/puppet/ssl/*
```

Comment:
This doesn't work at least for AIO packages and FreeBSD, maybe something like tapping `puppet config print ssldir` for the directory could work.